src/lib/__test__/use.px.test.ts

```diff
@@ -0,0 +1,78 @@
+import { Theme } from 'theme-ui'
+import { themeToPx } from '../use-px'
+
+describe('themeToPx', () => {
+  it('Should work if sizes is an array', () => {
+    const theme: Theme = {
+      sizes: [0, 4, 8],
+    }
+
+    const k = 1
+    const result = themeToPx(theme, k, 'sizes')
+    expect(result).toBe(theme.sizes![k])
+  })
+
+  it('Should work if sizes is a smaller array ', () => {
+    const theme: Theme = {
+      sizes: [0, 4, 8],
+    }
+
+    const k = 16
+    const result = themeToPx(theme, k, 'sizes')
+    expect(result).toBe('16px')
+  })
+
+  it('Should work if sizes a string array', () => {
+    const theme: Theme = {
+      sizes: [0, '4rem', '8rem'],
+    }
+
+    const k = 1
+    const result = themeToPx(theme, k, 'sizes')
+    expect(result).toBe(theme.sizes![k])
+  })
+
+  it('Should work if sizes is not defined', () => {
```

Comment:
if sizes key is not defined in theme, (I would leave an empty theme because at the beginning I didn't get what you were testing